sdk/core/azure-core/tests/testserver_tests/coretestserver/coretestserver/test_routes/streams.py

```diff
@@ -67,4 +66,16 @@ def compressed_no_header():
 @streams_api.route('/compressed', methods=['GET'])
 def compressed():
     return Response(stream_compressed_header_error(), status=300, headers={"Content-Encoding": "gzip"})
+
+def stream_decompress_header():
+    with gzip.open('test.tar.gz', 'wb') as f:
```

Comment:
Suggestion: use the [tempfile](https://docs.python.org/3/library/tempfile.html) module to help with cleanup etc.

Reply:
In fact, I am thinking of checking test.tar.gz into our tests so we don't need to generate it every time.

How do you think?

Reply:
Not writing files is goodness. Binary files checked in is less so. If the file is this small, then inlining the data would probably be simpler. We can also use `zlib` compress to compress data inline if we wanted to produce larger amounts of data.